src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -426,7 +426,7 @@ public void delete(long exerciseId, boolean deleteStudentReposBuildPlans, boolea
         tutorParticipationRepository.deleteAllByAssessedExerciseId(exercise.getId());
 
         if (exercise.isExamExercise()) {
-            Exam exam = examRepository.findOneWithEagerExercisesGroupsAndStudentExams(exercise.getExerciseGroup().getExam().getId());
+            Exam exam = examRepository.findOneWithEagerExercisesGroupsAndStudentExams(exercise.getExerciseGroup().getExam().getId()).orElse(null);
             for (StudentExam studentExam : exam.getStudentExams()) {
```

Comment:
Can the optional be empty and therefore `exam` actually be `null` at this point? If so, then `exam.getStudentExams()` would fail with an exception and the required clean-up for programming exercises below in lines 441ff would no longer run.